### PR TITLE
[Blaze] Parse and store `start_time` of a Blaze campaign list item

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #13477 

## Description

To schedule Blaze local notificafications we need to know the start time of a campaign. This PR starts parsing the start time of a Blaze campaign and stores it in core data. 

Changes
- Add a new core data version to add `startTime` attribute to `BlazeCampaignListItem`.
- Add migration tests.
- Parse the `start_time` field in `BlazeCampaignListItem` networking model.
- Include new `startTime` value while reading and writing `BlazeCampaignListItem` in storage. 
- Unit tests. 

## Steps to reproduce

- Install a build from the `trunk` (If you don't have Xcode, install build `20.2`)
- Login into a store eligible for Blaze
- Replace that build with a build from this branch (If you don't have Xcode install build `20.3`)
- Ensure that there is no crash on launch.
- Navigate to Menu -> Blaze
- Pull to refresh
- Ensure that campaigns are parsed and loaded successfully

## Testing information

- I tested that a previous build from `trunk` can be replaced with a build from this branch and launched without any crash.
- I tested that the start time field from the Blaze campaign list item response is parsed successfully. 

## Screenshots

Simulator Screen Recording - iPhone 15 Pro Max - 2024-09-02 at 11 05 41

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Networking/NetworkingTests/Responses/blaze-campaigns-list-success.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     {
       "id": "34518",
       "status": "rejected",
-      "start_time": "2024-01-05T00:00:00.000Z",
+      "start_time": "2024-09-02T04:58:03.000Z",
```
Comment (pmusolino) [inline Networking/NetworkingTests/Responses/blaze-campaigns-list-success.json:6]: It's not fully clear to me why this change 🤔

File: Storage/Storage/Model/WooCommerce.xcdatamodeld/.xccurrentversion [hunk 2]
```diff
@@ -3,6 +3,6 @@
 <plist version="1.0">
 <dict>
 	<key>_XCCurrentVersionName</key>
-	<string>Model 115.xcdatamodel</string>
+	<string>Model 116.xcdatamodel</string>
```
Comment (pmusolino) [inline Storage/Storage/Model/WooCommerce.xcdatamodeld/.xccurrentversion:6]: I noticed you didn't added this new model under `MIGRATIONS.md`, which is important to keep up to date 👍

Comment (pmusolino) [reply Storage/Storage/Model/WooCommerce.xcdatamodeld/.xccurrentversion:6]: I noticed that it's also missing in the testing method `test_it_can_load_the_expected_model_versions`

File: Storage/StorageTests/CoreData/MigrationTests.swift [hunk 3]
```diff
@@ -3017,6 +3017,34 @@ final class MigrationTests: XCTestCase {
...
+    func test_migrating_from_115_to_116_adds_new_startTime_attribute_to_BlazeCampaignListItem() throws {
+        // Given
+        let sourceContainer = try startPersistentContainer("Model 115")
+        let sourceContext = sourceContainer.viewContext
+
+        let campaign = insertBlazeCampaignListItem(to: sourceContext)
+        try sourceContext.save()
+
+        // Confidence check: new startTime attribute is not present
+        XCTAssertNil(campaign.entity.attributesByName["startTime"])
```
Comment (pmusolino) [inline Storage/StorageTests/CoreData/MigrationTests.swift:3032]: Just as a double check (because otherwise this can fake the test), it would be great to make sure that the inserted `BlazeCampaignListItem` has been saved. What do you think?

**Review by pmusolino — APPROVED** [review]
Looks good to me! I also tested that a previous build from `trunk` can be replaced with a build from this branch and launched without any crash.

Just left some minor nits 👍

File: Storage/Storage/Model/WooCommerce.xcdatamodeld/.xccurrentversion (hunk 2 above)
Comment (selanthiraiyan) [reply Storage/Storage/Model/WooCommerce.xcdatamodeld/.xccurrentversion:6]: 🤦 I added those in https://github.com/woocommerce/woocommerce-ios/pull/13807/commits/3fda68f1f1203b3f3bf79605b4e84b70b68dab5c and https://github.com/woocommerce/woocommerce-ios/pull/13807/commits/13c16d96184a1038ccc6dfb5af54f7383a376e5b.

File: Networking/NetworkingTests/Responses/blaze-campaigns-list-success.json (hunk 1 above)
Comment (selanthiraiyan) [reply Networking/NetworkingTests/Responses/blaze-campaigns-list-success.json:6]: I just updated the value to unit test the date [here](https://github.com/woocommerce/woocommerce-ios/pull/13807/files#diff-3240e7d366cbcc768e8ea2c53db05cc5f20c2a572e496c646f1f3d9e8743c548R33).

File: Storage/StorageTests/CoreData/MigrationTests.swift (hunk 3 above)
Comment (selanthiraiyan) [reply Storage/StorageTests/CoreData/MigrationTests.swift:3032]: Done in https://github.com/woocommerce/woocommerce-ios/pull/13807/commits/5d8263ddcee4ee8a58c7443f54b47e1b8c308989

**Comment by selanthiraiyan** [comment]
Thanks for the review, Paolo! 🙇 

I have pushed a few commits to address your comments and I am enabling auto-merge now. Please share if you have more comments.
